src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/TypeNamesShouldNotMatchNamespacesTests.cs

```diff
@@ -91,24 +80,30 @@ public class Forms
         }
 
         [Fact]
-        public void CA1724CSharpInvalidNameMatchingSdkNamespaceInDefaultRule()
+        public async Task CA1724CSharpInvalidNameMatchingSdkNamespaceInDefaultRule()
         {
-            var source = @"
+            await new VerifyCS.Test
+            {
+                TestCode = @"
 public class Sdk
 {
 }
-";
-            Document document = CreateDocument(source, LanguageNames.CSharp);
-            Project project = document.Project.AddMetadataReference(MetadataReference.CreateFromFile(typeof(Xunit.Sdk.AllException).Assembly.Location));
-            DiagnosticAnalyzer analyzer = GetCSharpDiagnosticAnalyzer();
-            GetSortedDiagnostics(analyzer, project.Documents.Single())
-                .Verify(analyzer, GetDefaultPath(LanguageNames.CSharp), CSharpDefaultResultAt(2, 14, "Sdk", "Xunit.Sdk"));
+",
+                SolutionTransforms =
+                {
+                    (solution, projectId) => solution.AddMetadataReference(projectId, MetadataReference.CreateFromFile(typeof(Xunit.Sdk.AllException).Assembly.Location))
+                },
```

Comment:
📝 You should be able to do this with the [`AdditionalReferences`](https://github.com/dotnet/roslyn-sdk/blob/180edd60921b738d2779b0e856974de0ff6b27f7/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ProjectState.cs#L31) property of the test state.

```
TestState =
{
  Sources = { ... },
  AdditionalReferences = { MetadataReference.CreateFromFile(typeof(Xunit.Sdk.AllException).Assembly.Location) },
}
```